### PR TITLE
set LazyModuleDir.value based on options or ini

### LINKED ISSUE #7: --tf-mod-dir value is not respected
When executing tests via either the `--tf-mod-dir` or setting the `terraform-mod-dir` neither updates the `tf.LazyModuleDir`

### PR DESCRIPTION
**Pull request by marcoceppi**
fixes #7

### CODE REVIEW

**Comment by marcoceppi** [comment]
I attempted to write a unit test for this but I found the process for writing unit tests for pytest modules itself really cumbersome and it didn't end up working. I omitted it from the branch to not break builds

```python
@pytest.mark.skipif(not tf.find_binary("terraform"), reason="Terraform binary missing")
def test_plugins_ini_setting_terraform_mod_dir(testdir):
    testdir.makeini(
        """
        [pytest]
        terraform-mod-dir = tf
    """
    )

    testdir.mkdir("tf")
    testdir.mkdir("tf/local_foobar")
    testdir.makefile(
        ".tf",
        foobar="""
        resource "local_file" "foobar" {
           content = "foobar!"
           filename = "${path.module}/foobar.txt"
        }
    """,
    )

    testdir.makepyfile(
        """
        import pytest
        from pytest_terraform import terraform

        @terraform("local_foobar")
        def test_local_foobar(local_foobar):
            assert local_baz['local_file.foobar.conent'] == 'foobar!'
            return
    """
    )

    result = testdir.runpytest("-v", "-s")

    # fnmatch_lines does an assertion internally
    result.stdout.fnmatch_lines(["*::test_local_foobar PASSED*"])

    # make sure that that we get a '0' exit code for the testsuite
    assert result.ret == 0
```

Decided to shelve it for now. Happy to revisit and get testing added for this as the line in question in `tf.py`

https://github.com/cloud-custodian/pytest-terraform/blob/ba5d71b3fa1388cb17695249ffbca08bd907e6db/pytest_terraform/tf.py#L258-L259

doesn't have coverage yet. I'd need to hunt around the pytest docs somemore to find how to patch parser options at runtime so I could just use a local_file test similar to `local_foo`.

**Comment by marcoceppi** [comment]
I found a way to test ini / cli

File: tests/test_terraform.py [hunk 1]
```diff
@@ -138,3 +139,34 @@ def test_hello_world(hello):
 
     # make sure that that we get a '0' exit code for the testsuite
     assert result.ret == 0
+
+
+@pytest.mark.skipif(not tf.find_binary("terraform"), reason="Terraform binary missing")
+def test_plugins_ini_setting_terraform_mod_dir(testdir):
+    mod_dir = Path(__file__).parent / "mrofarret"
```
Comment (kapilt) [inline tests/test_terraform.py:146]: nit. can you stick this in a data subdir off the test dir just to keep it obvious to purpose, plus we may have others.

Comment (marcoceppi) [reply tests/test_terraform.py]: can do!

**Pull request merged**